pkg/controller/argocd/hooks_test.go

```diff
@@ -0,0 +1,111 @@
+package argocd
+
+import (
+	"fmt"
+	"testing"
+
+	argoprojv1alpha1 "github.com/argoproj-labs/argocd-operator/pkg/apis/argoproj/v1alpha1"
+	"gotest.tools/assert"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/rbac/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+func testDeploymentHook(cr *argoprojv1alpha1.ArgoCD, v interface{}) error {
+	switch o := v.(type) {
+	case *appsv1.Deployment:
+		var replicas int32 = 3
+		o.Spec.Replicas = &replicas
+	}
+	return nil
+}
+
+func testClusterRoleHook(cr *argoprojv1alpha1.ArgoCD, v interface{}) error {
+	switch o := v.(type) {
+	case *v1.ClusterRole:
+		o.Rules = append(o.Rules, policyRuleForApplicationController()...)
+	}
+	return nil
+}
+
+func testErrorHook(cr *argoprojv1alpha1.ArgoCD, v interface{}) error {
+	switch o := v.(type) {
+	case *appsv1.Deployment:
+		err := fmt.Errorf("Error returned from %s for %s", "testErrorHook", o.Name)
+		return err
+	}
+	return nil
+}
+
+func TestReconcileArgoCD_testDeploymentHook(t *testing.T) {
+	defer resetHooks()()
+	logf.SetLogger(logf.ZapLogger(true))
+	a := makeTestArgoCDForClusterConfig()
```

Comment:
You could simplify this to just `makeTestArgoCD()` the tests don't require the cluster config.